velox/exec/TableWriter.cpp

```diff
@@ -249,6 +253,42 @@ void TableWriter::close() {
   }
 }
 
+void TableWriter::setConnectorOrWriterMemoryReclaimer(
+    memory::MemoryPool* pool) {
+  VELOX_CHECK_NOT_NULL(pool);
+  pool->setReclaimer(
+      TableWriter::MemoryReclaimer::create(operatorCtx_->driverCtx(), this));
+}
+
+std::unique_ptr<memory::MemoryReclaimer> TableWriter::MemoryReclaimer::create(
+    DriverCtx* driverCtx,
+    Operator* op) {
+  return std::unique_ptr<memory::MemoryReclaimer>(
+      new TableWriter::MemoryReclaimer(
+          driverCtx->driver->shared_from_this(), op));
+}
+
+bool TableWriter::MemoryReclaimer::reclaimableBytes(
+    const memory::MemoryPool& pool,
+    uint64_t& reclaimableBytes) const {
+  VELOX_CHECK(!pool.isLeaf());
+  return 0;
+}
+
+uint64_t TableWriter::MemoryReclaimer::reclaim(
+    memory::MemoryPool* pool,
+    uint64_t targetBytes) {
+  VELOX_CHECK(!pool->isLeaf());
+  return 0;
```

Comment:
Not calling parent's reclaim() method? Will it be connecting back to operator/driver's suspension logic?

Reply:
We shortcut the reclaim for connector memory pool